src/Core/Managed/Shared/Extensibility/Metrics/DependencyMetricExtractor.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using System.Collections.Generic;
+
+namespace Microsoft.ApplicationInsights.Extensibility.Metrics
+{
+    public class DependencyMetricExtractor : MetricExtractorTelemetryProcessorBase
+    {
+        private const string Version = "1.0";
+
+        private IDictionary<string, Metric> successDurationMetricsByType = null;
+        private IDictionary<string, Metric> failureDurationMetricsByType = null;
+
+        private Metric successDurationMetricDefault = null;
+        private Metric failureDurationMetricDefault = null;
+
+        private string[] dependencyTypes = null;
+        private string dependenccyTypesString = String.Empty;
```

Comment:
why do you need to keep both? For consistency it's better to only have one source of truce